gateway-discovery-cm/src/test/java/org/apache/knox/gateway/topology/discovery/cm/monitor/ServiceConfigurationRecordObjectMapperTest.java

```diff
@@ -0,0 +1,240 @@
+
+    assertEquals(clusterName, deserialized.getClusterName());
+    assertEquals(address, deserialized.getDiscoveryAddress());
+    Map<String, ServiceConfigurationModel> deserializedConfigs = deserialized.getConfigs();
+    assertNotNull(deserializedConfigs);
+    assertEquals(2, deserializedConfigs.size());
+    assertTrue(deserializedConfigs.containsKey("Service1"));
+    assertTrue(deserializedConfigs.containsKey("Service2"));
+
+    ServiceConfigurationModel service1 = deserializedConfigs.get("Service1");
+    Map<String, String> svc1Props = service1.getServiceProps();
+    assertEquals(2, svc1Props.size());
+    assertEquals(model1.getServiceProps().get("svc1.prop1"), svc1Props.get("svc1.prop1"));
+    assertEquals(model1.getServiceProps().get("svc1.prop2"), svc1Props.get("svc1.prop2"));
+    assertEquals(2, service1.getRoleTypes().size());
+    assertEquals(model1.getRoleProps("Role1").get("r1.prop1"), service1.getRoleProps("Role1").get("r1.prop1"));
+    assertEquals(model1.getRoleProps("Role1").get("r1.prop2"), service1.getRoleProps("Role1").get("r1.prop2"));
+    assertEquals(model1.getRoleProps("Role2").get("r2.prop1"), service1.getRoleProps("Role2").get("r2.prop1"));
+    assertEquals(model1.getRoleProps("Role2").get("r2.prop1"), service1.getRoleProps("Role2").get("r2.prop1"));
+
+
+    ServiceConfigurationModel service2 = deserializedConfigs.get("Service2");
+    Map<String, String> svc2Props = service2.getServiceProps();
+    assertEquals(1, svc2Props.size());
+    assertEquals(model2.getServiceProps().get("svc2.prop1"), svc2Props.get("svc2.prop1"));
+    assertEquals(2, service2.getRoleTypes().size());
+    assertEquals(model2.getRoleProps("Role1").get("r1.prop1"), service2.getRoleProps("Role1").get("r1.prop1"));
+    assertEquals(model2.getRoleProps("Role2").get("r2.prop1"), service2.getRoleProps("Role2").get("r2.prop1"));
+    assertEquals(model2.getRoleProps("Role2").get("r2.prop2"), service2.getRoleProps("Role2").get("r2.prop2"));
+  }
+
+  private String doTestSerialization(final ServiceConfigurationRecord record) {
+    String serialized = null;
+    try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
+      (new ServiceConfigurationRecordObjectMapper()).writeValue(out, record);
+      serialized = out.toString(StandardCharsets.UTF_8.name());
+    } catch (Exception e) {
+      fail(e.getMessage());
+    }
+    return serialized;
+  }
+
+  private ServiceConfigurationRecord doTestDeserialization(final String serialized) {
+    ServiceConfigurationRecord deserialized = null;
+    try (ByteArrayInputStream in = new ByteArrayInputStream(serialized.getBytes(StandardCharsets.UTF_8))) {
+      deserialized =
+          (new ServiceConfigurationRecordObjectMapper()).readValue(in, ServiceConfigurationRecord.class);
+    } catch (Exception e) {
+      fail(e.getMessage());
+    }
+    return deserialized;
+  }
+
+  @SuppressWarnings({"PMD.AppendCharacterWithChar", "PMD.ConsecutiveLiteralAppends"})
```

Comment:
Not sure why these are suppressed? It would be better to actually do what it says. One of the simple ones is use `'` around literals. ie: `'{'` or `'"'`